content/6_Advanced/LineContainer.mdx

```diff
@@ -30,13 +30,190 @@ frequency: 1
+
+### Incremental
+
+A slightly faster approach to this problem would be to incrementally construct the intersection of the half-planes one by one. We cut the convex polygon by a line $N$ amount of times, removing the redundant planes each time.
+
+If we represent the convex polygon as a set of line segments, we can cut it with a half-plane by finding the points of intersection of the segments with the half-plane line, replacing all the line segments in the middle with a new segment corresponding to a half-plane. This can be implemented in $\mathcal{O}(N)$ time, and decrease the bounding box with each half-plane, resulting in an $\mathcal{O}(N^2)$ complexity.
+
+This is much better than our initial approach. However, it's still relatively inefficient as we still have to loop over $\mathcal{O}(N)$ half-planes with every iteration. We will make some alterations to this algorithm in the next approach for a more reasonable solution.
+
+### Sort and Increment Algorithm
+
+We can use some properties of the output of the previous algorithm to optimize it. We know that the resulting region of the intersection of half-planes is convex, therefore it will consist of some segments of half-planes in order of their angles. In other words, if we incrementally intersect the half-planes in the order of their angles, storing them in a double-ended queue, we will only need to remove half-planes from the front and the back of the queue.
+
+Let's say that the half-plane angles are sorted from $-π$ to $π$ and that we are about to start the $k$-th step of the algorithm. We can say that we have already constructed the intersection of the half-planes up to $k-1$. Because the half-planes are sorted by angle, we can be sure that $k$ will form a convex turn with the $k-1$ half-plane. From this, we can say:
+* Some of the half-planes that are located at the back of the intersection may become redundant, in which case we shall pop from the queue from the back of the double-ended queue.
+* Some of the half-planes that are located at the front of the intersection may become redundant, in which case we shall pop from the queue from the back of the double-ended queue.
+* The intersection may become empty from this, in which case we shall simply terminate the algorithm and return the intersection as empty.
+
+In this case, redundant means that it doesn't change the intersection, or the half-plane could be removed and the intersection wouldn't change.
+
+To visualize this, we shall let $H = A,B,C,D,E$. This is the set of half-planes that are currently present in the intersection. For the points, we can define them as $P = p,q,r,s$ which are intersection points of adjacent half-planes from $H$. This could be represented as such below:
+
+![image2](halfplane2.png)
```

Comment:
make sure that you import it under the images folder in 6_Advanced.
![image](https://github.com/cpinitiative/usaco-guide/assets/81596991/9afe6c80-5f46-4b47-8161-711579322b0d)

Reply:
the images still aren't loading for some reason even though i added them into the 6_Advanced folder and here in the files in this pr.

Reply:
nvm, i figured it out. the images are loading now.